### ISSUE TITLE
Enhancement: [key-spacing] also apply on function arguments

### ISSUE THREAD

**Issue by coyotte508** [typescript-eslint/typescript-eslint#6408]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/key-spacing/

### Description

I propose that the rule should also apply on spaces between argument names and argument types.

### Fail

```typescript
// with {align: "value"}
function f(
  a: number,
  abc: string
)
```

### Pass

```typescript
// with {align: "value"}
function f(
  a:   number,
  abc: string
)
```
